Products/Rider.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Rider plugins
+---
+
+
+Rider is a cross-platform .NET IDE. While all language-related features discussed in this guide can by run by Rider in a ReSharper background process, the actual IDE frontend is based on the IntelliJ platform. The ReSharper backend is either running on the .NET Framework or Mono, depending on installations and operating system. The frontend, however, is running on the JVM. In order to make those two process talk with each other and let them share data, some kind of protocol is required. For instance, the frontend might need to tell the ReSharper process that we hit <kbd>Tab</kbd> after a `value.`, and that auto-completion items should be calculated. In return, the background process needs to share all the possible items with the frontend, so that the user can chose from a list. This protocol is implemented using [Kotlin](https://kotlinlang.org/).
+
+As an author of a ReSharper extension, we might want to make it available for Rider as well. Depending on the types and extension points we are using, this sometimes doesn't require so much additional work, but only to create a Rider [plugin project]() and to follow Rider's [plugin packaging](). This is usually true if we:
+
+- Don't use any XAML option pages or tool windows
+- Don't use any VisualStudio components
+
+In all other cases, we would need to re-implement missing parts using the [IntelliJ Platform SDK](http://www.jetbrains.org/intellij/sdk/docs/welcome.html), which is structured similar to this guide. For interprocess communication, we can easily [extend the protocol]() using a Kotlin DSL.
+
+## Plugin project (.NET)
+
+The standard approach of migrating an existing ReSharper plugin to also work for Rider, is to copy the existing CSPROJ file and adapt the package reference from `JetBrains.ReSharper.SDK` to `JetBrains.Rider.SDK`.
+
+> **NOTE:** It is strongly recommended to use SDK-based project files to decrease friction.
```

Comment:
What are SDK-based project files? (I know, but readers may ask themselves... maybe add a link to MSDN)